src/codegen/generate.ts

```diff
@@ -283,6 +284,9 @@ export default class ApiGenerator {
     public readonly isConverted = false,
   ) {}
 
+  // Store parent schemas (see `preprocessComponents` for the definition of a parent schema)
+  parentSchemas: Set<string> = new Set();
```

Comment:
Would still like to give this one a name that's more clear towards why this exists. I feel like `parent` is too generic.

Can we use `discriminatingSchemas` maybe?

Reply:
nice name!